client/src/infographic/Infographic.js

```diff
@@ -88,67 +88,26 @@ class InfoGraph_ extends React.Component {
       ["subject", "active_bubble_id", "level"]
     );
 
-    const should_reload_panel_dependencies =
-      !prevState.bubble_menu_loading && prevState.infographic_loading;
-
     if (should_reload_bubble_menu_dependencies) {
       return {
-        bubble_menu_loading: true,
-        infographic_loading: true,
+        loading: true,
         subject: nextProps.subject,
         subject_panels_by_bubble_id: {},
         active_bubble_id: nextProps.active_bubble_id,
         level: nextProps.level,
       };
-    } else if (should_reload_panel_dependencies) {
-      const { subject_panels_by_bubble_id, subject } = prevState;
-      const { active_bubble_id } = nextProps;
-
-      const subject_bubble_defs = _.chain(bubble_defs)
-        .filter(({ id }) =>
-          _.chain(subject_panels_by_bubble_id).keys().includes(id).value()
-        )
-        .map((bubble_def) =>
-          _.mapValues(bubble_def, (bubble_option) =>
-            _.isFunction(bubble_option) ? bubble_option(subject) : bubble_option
-          )
-        )
-        .value();
-
-      const active_index = _.findIndex(subject_bubble_defs, {
-        id: active_bubble_id,
-      });
-      const next_bubble = subject_bubble_defs[active_index + 1];
-      const previous_bubble = subject_bubble_defs[active_index - 1];
-
-      const panel_keys = subject_panels_by_bubble_id[active_bubble_id];
-
-      return {
-        subject_bubble_defs,
-        previous_bubble,
-        next_bubble,
-        panel_keys,
-        infographic_loading: true,
-      };
     } else {
       return null;
     }
   }
   componentDidMount() {
-    this.loadBubbleMenuDeps(this.props);
+    this.loadBubbleMenuAndGraphDeps(this.props);
```

Comment:
So, I had to read this over a bit to refresh my memory. Essentially, `loadGraphDeps` need `panel_keys_by_bubble_id` which is loaded from `loadBubbleMenuDeps`. With that in mind, I didn't see much need to keep 2 loads separated, hence combined the two. Performance doesn't seem much impacted but let me know if it seems to be a problem

Reply:
Yeah, one had to happen after another which was managed through the two different loading flags on the component state/calls in `componentDidUpdate`. Cleaning up that state slightly's not bad, but I think keeping the two loading steps clearly separate is useful for readability (since it really is two different loading stages). It was also slightly more flexible towards future changes when they were distinct (maybe at some point we decide there's something else we want to render in-between loading bubble menu deps and graph deps?).

Mostly just my own opinion, but I'd still prefer keeping them distinct. Could hit a middle ground where there's still only one flag in the state and `componentDidMount` calls `this.loadBubbleMenuDeps(this.props).then( () => this.loadGraphDeps(this.props)) ` or something, with only graph deps making a state change.

Reply:
Okay, kind of met it in the middle road as how you mentioned. Can't separate loading state though